### ISSUE TITLE
Can I use clip and round in lambda layer and merge layer?

### ISSUE THREAD

**Issue by liujxing** [fchollet/keras#5107]
I am using Keras with Tensorflow backend, and I intended to clip and round the output of an intermediate layer with code like output = Lambda(lambda x: K.round(K.clip(x))(input), but this code will raise the value error "None type not supported" from Tensorflow. Does that mean I should not use clip or round in such layers as they produce zero gradient almost everywhere?

**Comment by bstriner** [comment]
@liujxing You might be getting the error due to the shape calculation, and not the lambda itself. Try adding `output_shape=lambda x:x` to `Lambda` and let me know if you sill have the issue.

Why would you want to be using round? Rounding is a step-wise function so training your network won't do anything past the rounding layer. Just keep in mind that gradients don't propagate through the rounding. Same is true of clipping if the values are clipped.

Cheers

**Comment by bstriner** [comment]
Also, safer to pass min and max to clip. Not sure what the defaults are.
